x-pack/plugins/stack_connectors/public/connector_types/cases_webhook/steps/get.tsx

```diff
@@ -6,103 +6,187 @@
 const { emptyField, urlField } = fieldValidators;
 
 interface Props {
   display: boolean;
   readOnly: boolean;
 }
 
-export const GetStep: FunctionComponent<Props> = ({ display, readOnly }) => (
-  <span data-test-subj="getStep" style={{ display: display ? 'block' : 'none' }}>
-    <EuiText>
-      <h3>{i18n.STEP_3}</h3>
-      <small>
-        <p>{i18n.STEP_3_DESCRIPTION}</p>
-      </small>
-    </EuiText>
-    <EuiFlexGroup direction="column">
-      <EuiFlexItem>
-        <UseField
-          path="config.getIncidentUrl"
-          config={{
-            label: i18n.GET_INCIDENT_URL,
-            validations: [
-              {
-                validator: urlField(i18n.GET_INCIDENT_URL_REQUIRED),
-              },
-              { validator: containsExternalId() },
-            ],
-            helpText: i18n.GET_INCIDENT_URL_HELP,
-          }}
-          component={MustacheTextFieldWrapper}
-          componentProps={{
-            euiFieldProps: {
-              readOnly,
-              'data-test-subj': 'webhookGetUrlText',
-              messageVariables: urlVars,
-              paramsProperty: 'getIncidentUrl',
-              buttonTitle: i18n.ADD_CASES_VARIABLE,
-              showButtonTitle: true,
-            },
-          }}
-        />
-      </EuiFlexItem>
-      <EuiFlexItem>
-        <UseField
-          path="config.getIncidentResponseExternalTitleKey"
-          config={{
-            label: i18n.GET_INCIDENT_TITLE_KEY,
-            validations: [
-              {
-                validator: emptyField(i18n.GET_RESPONSE_EXTERNAL_TITLE_KEY_REQUIRED),
+export const GetStep: FunctionComponent<Props> = ({ display, readOnly }) => {
+  const [{ config }] = useFormData({
+    watch: ['config.getIncidentMethod'],
+  });
+  const { getIncidentMethod = WebhookMethods.GET } = config ?? {};
+
+  return (
+    <span data-test-subj="getStep" style={{ display: display ? 'block' : 'none' }}>
+      <EuiText>
+        <h3>{i18n.STEP_3}</h3>
+        <small>
+          <p>{i18n.STEP_3_DESCRIPTION}</p>
+        </small>
+      </EuiText>
+      <EuiFlexGroup direction="column">
+        <EuiFlexGroup justifyContent="spaceBetween">
+          <EuiFlexItem grow={false}>
+            <UseField
+              path="config.getIncidentMethod"
+              component={Field}
+              config={{
+                label: i18n.GET_INCIDENT_METHOD,
+                defaultValue: 'get',
+                type: FIELD_TYPES.SELECT,
+                validations: [
+                  {
+                    validator: emptyField(i18n.GET_METHOD_REQUIRED),
+                  },
+                ],
+              }}
+              css={css`
+                .euiFormRow__labelWrapper {
+                  margin-bottom: 9px;
+                }
+              `}
+              componentProps={{
+                euiFieldProps: {
+                  'data-test-subj': 'webhookGetIncidentMethodSelect',
+                  options: HTTP_VERBS.map((verb) => ({
```

Comment:
I believe `PUT` and `PATCH` are only used in the Create and Update steps, so we don't want to map all 4 `HTTP_VERBS` here. Just `GET` and `POST`.

Reply:
[updated](https://github.com/elastic/kibana/pull/197437/commits/5eb601964137d0dc47db0640f312914da19a91cf)